beta-src/src/components/ui/WDCountdownPill.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from "react";
+import { useEffect, useState } from "react";
+import { Avatar, Chip } from "@mui/material";
+import BlackCountdownIcon from "../../assets/png/icn_phase_countdown_black.png";
+import RedCountdownIcon from "../../assets/png/icn_phase_countdown_red.png";
+import calculateAndSortRemainingTime from "../../utils/calculateAndSortRemainingTime";
+import { ParsedTimeInterface } from "../../interfaces/ParsedTimeInterface";
+
+interface WDCountdownPillProps {
+  remainingTime: number;
+}
+
+const WDCountdownPill: React.FC<WDCountdownPillProps> = function ({
+  remainingTime,
+}) {
+  const [endTime] = useState<number>(+new Date(remainingTime * 1000));
+
+  const [quarterTimeRemaining] = useState(
```

Comment:
same question...just use a variable

Reply:
This is used as state so that its value does not keep getting called every time the setTimeLeft is called since its value is dependent on new Date being called. new Date's value is needed to be able to calculate the difference between now and end game time but shouldnt be repeating for this specific value.